### ISSUE TITLE
[K-Bug] `left` and `right` assoc parsing symbols should not be allowed on non-assoc productions

### ISSUE THREAD

**Issue by ehildenb** [runtimeverification/k#3415]
### What component is the issue in?

Front-End

### Which command

- [X] kompile
- [ ] kast
- [ ] krun
- [ ] kprove
- [ ] kprovex
- [ ] ksearch

### What K Version?

K version:    v5.6.91

### Operating System

Linux

### K Definitions (If Possible)

```
module TEST
    imports BOOL
    imports INT
    
    configuration <k> $PGM:Pgm </k>

    syntax Pgm ::= Bool | Int

    syntax Bool ::= Int "pred" Int [left, function]

    rule I1 pred I2 => I1 <=Int I2

endmodule
```

### Steps to Reproduce

```
% kompile test.k
[Warning] Compiler: Could not find main syntax module with name TEST-SYNTAX in
definition.  Use --syntax-module to specify one. Using TEST as default.
```

### Expected Results

I expect a kompile-time error saying: "`left` attribute not allowed on non-associative production `_pred_TEST_Int_Int_Bool`" or something similar.

See: https://github.com/runtimeverification/evm-semantics/pull/1726/files#r1194084829

**Comment by Baltoli** [comment]
Check that the sorts line up appropriately wrt. subsorting - return sort should be <= the argument sorts.

@radumereuta has an example for this

**Comment by radumereuta** [comment]
The priority and associativity filter is implemented [here](https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/parser/inner/disambiguation/PriorityVisitor.java#L179).
The idea behind priorities and assoc is to reject ASTs where the parent-child relation is not permitted.

`* > +` means `*` is not allowed as a direct child of `+`
`left *` means `*` is not allowed as a direct right most child of `*`.

So a sanity check for productions should look for
`syntax A ::= B "+" C`
- left - A <= C
- right - A <= B
- non-assoc - both of the above.

Probably the best place to put this check is [here](https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/kompile/Kompile.java#L422)

**Comment by Scott-Guest** [comment]
Closed by #3437
